### ISSUE TITLE
Interpolation of quoted strings differs from Ruby SASS

### ISSUE THREAD

**Issue by wtfribley** [sass/libsass#694]
Given following (albeit wonky) SASS

```css
$str: '{' + '"foo": "bar"' + '}';

.interpolation-test {
  test: "#{$str}";
}
```

When compiled using Ruby:

```css
.interpolation-test {
  test: '{"foo": "bar"}';
}
```

The quoting behavior is different in libsass:

```css
.interpolation-test {
  test: "{"foo": "bar"}";
}
```

While this example may be odd, I think perhaps taking a look at the behavior of quotes when interpolating strings may not be a bad idea (looks like it's come up before in #323).

**Comment by HugoGiraudel** [comment]
Possibly related: https://github.com/sass/libsass/issues/615.

**Comment by xzyfer** [comment]
Not really #615. 

#615 is related to parsing interpolants, this is about preserving quotes styles.

**Comment by mgreter** [comment]
The logic here seems pretty simple. If the string contains double quotes and no single quotes, it is wrapped in single quotes. This seems to be true for all output styles (I checked).

By the way, I used this test to verify the behaviour:

```css
$str01: "{" + '"' + "}";
$str02: '{' + '"' + '}';
$str03: '{' + '"' + "}";
$str04: "{" + '"' + '}';

$str05: "{" + "\"" + "}";
$str06: '{' + "\"" + '}';
$str07: '{' + "\"" + "}";
$str08: "{" + "\"" + '}';

$str11: "{" + "'" + "}";
$str12: '{' + "'" + '}';
$str13: '{' + "'" + "}";
$str14: "{" + "'" + '}';

$str15: "{" + '\'' + "}";
$str16: '{' + '\'' + '}';
$str17: '{' + '\'' + "}";
$str18: "{" + '\'' + '}';

$str21: "{" + "\"'" + "}";
$str22: '{' + "\"'" + '}';
$str23: '{' + "\"'" + "}";
$str24: "{" + "\"'" + '}';

$str25: "{" + '\'"' + "}";
$str26: '{' + '\'"' + '}';
$str27: '{' + '\'"' + "}";
$str28: "{" + '\'"' + '}';

.interpolation-test {
  test01: "#{$str01}";
  test02: "#{$str02}";
  test03: "#{$str03}";
  test04: "#{$str04}";
  test05: "#{$str05}";
  test06: "#{$str06}";
  test07: "#{$str07}";
  test08: "#{$str08}";

  test11: "#{$str11}";
  test12: "#{$str12}";
  test13: "#{$str13}";
  test14: "#{$str14}";
  test15: "#{$str15}";
  test16: "#{$str16}";
  test17: "#{$str17}";
  test18: "#{$str18}";

  test21: "#{$str21}";
  test22: "#{$str22}";
  test23: "#{$str23}";
  test24: "#{$str24}";
  test25: "#{$str25}";
  test26: "#{$str26}";
  test27: "#{$str27}";
  test28: "#{$str28}";
}
```

Results in
```css
.interpolation-test {
  test01: '{"}';
  test02: '{"}';
  test03: '{"}';
  test04: '{"}';
  test05: '{"}';
  test06: '{"}';
  test07: '{"}';
  test08: '{"}';
  test11: "{'}";
  test12: "{'}";
  test13: "{'}";
  test14: "{'}";
  test15: "{'}";
  test16: "{'}";
  test17: "{'}";
  test18: "{'}";
  test21: "{\"'}";
  test22: "{\"'}";
  test23: "{\"'}";
  test24: "{\"'}";
  test25: "{'\"}";
  test26: "{'\"}";
  test27: "{'\"}";
  test28: "{'\"}";
}
```

**Comment by P233** [comment]
Interpolation handles quoted string list incorrectly. The following code

```scss
$a: ".class-a", ".class-b";
$b: ".class-a" ".class-b";

#{$a} {
  display: none;
}

#{$b} {
  display: none;
}
```
is compiled to:

```css
.class-",".class-b {
  display: none; }

.class-"".class-b {
  display: none; }
```

**Comment by xzyfer** [comment]
@P233 that's fixed and will be 3.2.0 https://github.com/sass/libsass/issues/639

**Comment by xzyfer** [comment]
Want to update the spec @mgreter https://github.com/sass/sass-spec/tree/master/spec/libsass-todo-issues/issue_694?

**Comment by mgreter** [comment]
This should be solved by #910!
